Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts

```diff
@@ -563,7 +563,8 @@ export class ResourceGroup {
                                     setVariablesInObject(`${path}.${key}`, obj[key]);
                                 }
                                 else {
-                                    console.log(`##vso[task.setvariable variable=${path}.${key};]` + JSON.stringify(obj[key]));
+                                    const value = Array.isArray(obj[key]) ? JSON.stringify(obj[key]) : obj[key];
```

Comment:
Could you please give us more insights for the cases that are not "array" ?
I'm trying to understand why you've limited  `JSON.stringify` operation only for `Array.isArray`.

Reply:
the whole point of this PR was that it just outputs the actual value instead of a stringified version including quotes.

According to the [docs on variables](https://learn.microsoft.com/en-us/azure/devops/pipelines/process/set-variables-scripts?view=azure-devops&tabs=bash#about-tasksetvariable) it should output `##vso[task.setvariable variable=myVar;]foo` for a variable called `myVar` with a value `foo`. The current implementation (with stringify) outputs `##vso[task.setvariable variable=myVar;]"foo"` that means the value of that variable will be `"foo"`. And because it's `"foo"` it cannot be used in other steps and it's up to the user to parse the entire object.

The only time JSON.stringify is valid is for Arrays, then the value of the output should be `["value1", "value2"]` which is exactly what stringify does.

Reply:
Oh; I see your point. However; since this quote behavior exists for a long time; there might be some users who are already use variables with `quotes`.

If we change behavior suddenly; these users might be affected. Let me discuss this topic with our PM.

Reply:
Mind that the old behavior **is not** in the current available documentation. I only found out by looking through the code. Probably no one is using it.

Reply:
@svrooij; We've a conclusion for not disturbing existing behavior and your functionality.

Let's add a simple checkbox here:
<img width="882" alt="image" src="https://user-images.githubusercontent.com/2474419/226319553-5fb91dda-eea4-47c0-80d7-a56bb9464648.png">

With this explanation by default unchecking:
`[ ] Use value as it is without converting JSON`

If it is checked => it will use your change
if it is not checked => it will use existing mechanism

Do you know how to add a new input? Or we can also modify this PR accordingly if you're ok with changes in your source branch?

P.s. : we'll keep documentation's old version for "unchecked" behaviour and add your example as "checked" behavior.

Reply:
I have no clue on how to add inputs. I worked on this a long time ago 😄.

How about calling it `Stringify outputs using JSON.strinify (legacy)` and interpret not specified as `true`. Because it does something legacy which is considered incorrect. So `empty == true`, `true == true`, `false == false`. This could be checked with:

```js
if (inputs.legacyOutputs === false) {
// do my implementation
} else {
// do original (legacy) implementation
}
```

Then if people use the graphical interface the default value is unchecked and saved as such `false`. And people already using it can keep using the old behavior.